library/src/test/java/com/mapzen/valhalla/RouterTest.java

```diff
@@ -160,10 +160,10 @@ public void shouldAddLocations() throws Exception {
                 .setLocation(loc1)
                 .setLocation(loc2)
                 .getJSONRequest();
-        assertThat(json.locations[0].lat).contains("1.0");
-        assertThat(json.locations[0].lon).contains("2.0");
-        assertThat(json.locations[1].lat).contains("3.0");
-        assertThat(json.locations[1].lon).contains("4.0");
+        assertThat(json.locations.get(0).lat).contains("1.0");
+        assertThat(json.locations.get(0).lon).contains("2.0");
+        assertThat(json.locations.get(1).lat).contains("3.0");
+        assertThat(json.locations.get(1).lon).contains("4.0");
```

Comment:
There should be a unit test added to exercise constructing a request with more than two locations. What do you think about this?

```kt
    @Test
    public void shouldAddWaypoints() throws Exception {
        double[] loc1 = { 1.0, 2.0 };
        double[] loc2 = { 3.0, 4.0 };
        double[] loc3 = { 5.0, 6.0 };
        double[] loc4 = { 7.0, 8.0 };
        JSON json = new ValhallaRouter()
            .setLocation(loc1)
            .setLocation(loc2)
            .setLocation(loc3)
            .setLocation(loc4)
            .getJSONRequest();
        assertThat(json.locations.get(0).lat).contains("1.0");
        assertThat(json.locations.get(0).lon).contains("2.0");
        assertThat(json.locations.get(1).lat).contains("3.0");
        assertThat(json.locations.get(1).lon).contains("4.0");
        assertThat(json.locations.get(2).lat).contains("5.0");
        assertThat(json.locations.get(2).lon).contains("6.0");
        assertThat(json.locations.get(3).lat).contains("7.0");
        assertThat(json.locations.get(3).lon).contains("8.0");
    }
```